pymc/distributions/discrete.py

```diff
@@ -1224,7 +1226,16 @@ class Categorical(Discrete):
 
     @classmethod
     def dist(cls, p, **kwargs):
-
+        if isinstance(p, np.ndarray) or isinstance(p, list):
+            if (np.array(p) < 0).any():
```

Comment:
```suggestion
            if (np.asarray(p) < 0).any():
```
This is slightly more efficient if the input is already a numpy array